### ISSUE TITLE
Unable to Create Project: error while waiting for branch main to reach 'protected' status

### ISSUE THREAD

**Issue by tschechniker** [gitlabhq/terraform-provider-gitlab#1127]
### GitLab Provider version

3.15.1

### GitLab version

14.10.3

### Terraform version

1.1.3

### Relevant Terraform Configuration

```hcl
resource "gitlab_project" "test_project" {
    name = "test"
    description = "test"
    default_branch = "main"
    initialize_with_readme = true
}
```

### Relevant log output

```shell
Error: error while waiting for branch main to reach 'protected' status, timeout while waiting for state to become 'true'
```

### Description

Currently i'm unable to create a new Project if it's not in a group where branch protection is turned off in the group settings. It seems like the default behaviour of gitlab has changed and in the project creation there is no default branch protection any more. This also applies if i create the project manually. There is not branch protection created for the default branch.

https://github.com/gitlabhq/terraform-provider-gitlab/blob/main/internal/provider/resource_gitlab_project.go#L1083 here you rely on that there will always be a branch protection. And course of the fact that gitlab will not create one the timeout of 2 minutes hit always.

Till now i was unable to find any working workaround for this issue. Am i the only one?

**Comment by timofurrer** [comment]
Thanks for the report @tschechniker 🎉 

Interesting, we've a coupled of these issues lately also related to the `initialize_with_readme = true`, see #1111. If you set it to `false`, are you experiencing the same issue?

> [...] here you rely on that there will always be a branch protection.

Not really *always* as for groups for which it is turned of, we don't rely on that, see:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/dbc86a65b8268761ee28211036a8e5b01a44b79f/internal/provider/resource_gitlab_project.go#L1761

What I'm wondering is what your `default_branch_protection` on instance level is? Can you [verify that](https://docs.gitlab.com/ee/api/settings.html#get-current-application-settings)? Did this change for your instance?

**Comment by tschechniker** [comment]
if i turn initialize_with_readme to off it works yes but this will break all other steps so no option for me currently. 

I will check what the instance level branch protection is set to.

**Comment by timofurrer** [comment]
> if i turn initialize_with_readme to off it works yes but this will break all other steps so no option for me currently.

Alright, yes, was just out of curiosity and not meant to be a workaround :)

> I will check what the instance level branch protection is set to.

Thanks, keep as posted.

I'll draft a PR to also check the application defaults, since that's a bug IMHO anyways ...

**Comment by timofurrer** [comment]
another interesting thing would be to know what the instance-wide `default_branch_name` is?

**Comment by tschechniker** [comment]
@timofurrer the terraform module is not trying to delete and recreate the default_branch so i assume it's set to main but i will verify this too

**Comment by timofurrer** [comment]
@tschechniker reading this comment:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/dbc86a65b8268761ee28211036a8e5b01a44b79f/internal/provider/resource_gitlab_project.go#L1085

I assume that the instance-wide default branch name is not `main`, so that `default_branch = "main"` in your tf config becomes a *custom default branch*. At the same time probably `default_branch_protection` on instance level is disabled which leads to the issue at hand.

(given that the comment is actually true 😅 )

**Comment by tschechniker** [comment]
Okay i can confirm the instance wide branch is main and the default protection is set to disabled. So that gitlab will not create a default protection on project creation. 

I'm not sure why this should only affect when it's a custom branch name. Actually i would think if the set a custom branch name the logic in the module would be kind of self healing and add the branch protection to the new branch even if the original default branch did not have one.

**Comment by tschechniker** [comment]
Actually i retested all of your scenarios. All of those three does not work. I receive the same timeout issue

**Comment by timofurrer** [comment]
Okay, pretty strange ...

Do you have a way to test this branch for you: https://github.com/gitlabhq/terraform-provider-gitlab/pull/1128 and see if it works? At least the test in the pipeline seem to work - which they didn't before the change in the resource itself ...

**Comment by PatrickRice-KSC** [comment]
@tschechniker this issue wound up getting closed because the #1128 fix was included in the next milestone, but did you get a chance to confirm that this resolved your issue?  I just noticed we didn't get confirmation when I was reviewing release notes 🙁
